image_transport/CMakeLists.txt

```diff
@@ -58,11 +59,33 @@ target_link_libraries(list_transports
   ${PROJECT_NAME}
   pluginlib::pluginlib)
 
-# Build republish
-add_executable(republish src/republish.cpp)
+# # Build republish
+add_library(republish_node SHARED
+  src/republish.cpp
+)
+target_link_libraries(republish_node
+  ${PROJECT_NAME}
+)
+ament_target_dependencies(republish_node
+  pluginlib
+  rclcpp_components
+  rclcpp
+)
+rclcpp_components_register_nodes(republish_node "image_transport::Republisher")
```

Comment:
You could actually remove the entire add_executable below, delete republish_program.cpp and the utilities files and generate a node that loads the plugin and runs it with:

```
rclcpp_components_register_node(republish_node
  PLUGIN "image_transport::Republisher"
  EXECUTABLE republish
)
```